### ISSUE TITLE
CORS errors when using Gutenberg editor

### ISSUE THREAD

**Issue by drewbaker** [funkhaus/fuxt-backend#5]
When using Gutennberg editor, the console logs a lot of errors:

<img width="1438" alt="Screen Shot 2021-04-12 at 2 35 03 PM" src="https://user-images.githubusercontent.com/475015/114395089-4765f600-9b9c-11eb-98aa-7d956cacde31.png">

I tracked it down to this function: https://github.com/funkhaus/fuxt-backend/blob/master/functions/wp-functions.php#L280-L284

So now the question is, how to fix it? It seems weird that Gutenberg is doing API requests to the home URL, not the actual WordPress install URLs.

The whole point of that code was to get preview URL's linking out to the frontend correctly. So perhaps it's OK to just do a check for "is REST request" and if so, return the correct default home URL?

**Comment by drewbaker** [comment]
@rsm0128 So your PR:

```
    /*
     * Return the fuxt_home_url value when code requests the Site Address (URL)
     */
    function fuxt_get_home_url( $url, $path, $orig_scheme ) {
        if ( 'rest' !== $orig_scheme ) {
            $fuxt_home_url = get_option( 'fuxt_home_url ' );
            if ( ! empty( $fuxt_home_url ) ) {
                $url = $fuxt_home_url;
            }
        }
        return $url;
    }
    //add_filter('home_url', 'fuxt_get_home_url', 10, 3);
```

Works, but then WP-GQL (and WordPress) has all the post and pages URL's be `/`. So I think we got to think of a different solution.

**Comment by drewbaker** [comment]
If I disable that function, then resave the permalinks, the URL's all come back to what they should be.

**Comment by drewbaker** [comment]
Fixed with #7
